source/log_formatter.c

```diff
@@ -109,16 +109,16 @@ int aws_format_standard_log_line(struct aws_logging_standard_formatting_data *fo
         /*
          * Add thread id and user content separator (" - ")
          */
-        uint64_t current_thread_id = aws_thread_current_thread_id();
+        aws_thread_id current_thread_id = aws_thread_current_thread_id();
+        char repr[AWS_THREAD_ID_REPR_LEN];
+        if (aws_thread_id_to_string(current_thread_id, repr, AWS_THREAD_ID_REPR_LEN)) {
```

Comment:
The logger is pretty slow naturally, but I'm still a little worried about potential overhead here for Trace situations.  The value is const, so what about adding the thread_id's string value as a member of aws_thread and a getter for that value rather than doing the to-string loop with every log call?

Reply:
Yes, I see... except, I think this wouldn't work for logging in the main procedure since that isn't tied to an aws_thread instance. Instead, I made a thread-local copy of the string representation so the cost is at most once per thread.